Cython/Build/Inline.py

```diff
@@ -109,19 +109,175 @@ def _get_build_extension():
     return build_extension
 
 
+def cython_inline_raw(
+        code,
+        preprocessing_fn=None,
```

Comment:
It seems taking preprocessing_fn and transform_code_fn as deferred callbacks makes this code harder to follow. Is the intent to see if we hit the cache before we apply these functions? If so, could we just apply both of these before invoking cython_inline_raw (possibly requiring a separate cache on the unprocesssed code, which I think should be fine.)